packages/vm/src/context.rs

```diff
@@ -111,11 +120,22 @@ pub(crate) fn move_out_of_context<S: Storage, Q: Querier>(
 /// Moves owned instances of storage and querier into the context.
 /// Should be followed by exactly one call to move_out_of_context when the instance is finished.
 pub(crate) fn move_into_context<S: Storage, Q: Querier>(target: &mut Ctx, storage: S, querier: Q) {
-    let b = get_context_data::<S, Q>(target);
+    let b = get_context_data_mut::<S, Q>(target);
     b.storage = Some(storage);
     b.querier = Some(querier);
 }
 
+/// Returns true iff the storage is set to readonly mode
+pub fn is_storage_readonly<S: Storage, Q: Querier>(ctx: &Ctx) -> bool {
+    let context_data = get_context_data::<S, Q>(ctx);
+    context_data.storage_readonly
+}
+
+pub fn set_storage_readonly<S: Storage, Q: Querier>(ctx: &mut Ctx, new_value: bool) {
+    let mut context_data = get_context_data_mut::<S, Q>(ctx);
+    context_data.storage_readonly = new_value;
+}
+
```

Comment:
Both these methods can be shortened into one line each, but i understand if you wrote them like this intentionally

Reply:
Yeah, I prefer to use `get_context_data_mut` as in the other places. Not also no strong opinion here